Common/include/containers/CLookUpTable.hpp

```diff
@@ -0,0 +1,152 @@
+/*!
+ * \file CLookupTable.hpp
+ * \brief tabulation of fluid properties
+ * \author D. Mayer, T. Economon
+ * \version 7.3.1 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2022, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include <fstream>
+#include <iomanip>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "../../Common/include/option_structure.hpp"
+#include "../../Common/include/linear_algebra/blas_structure.hpp"
+#include "../../Common/include/toolboxes/CSquareMatrixCM.hpp"
+#include "CFileReaderLUT.hpp"
+#include "CTrapezoidalMap.hpp"
+
+using namespace std;
+
+class CLookUpTable {
+ protected:
+  int rank; /*!< \brief MPI Rank. */
+
+  string file_name_lut;
+  string type_lut;
+  string version_lut;
+  string version_reader;
+  unsigned long n_points;
+  unsigned long n_triangles;
+  unsigned long n_variables;
+  unsigned long n_hull_points;
+
+  su2double limits_table_enth[2];
+  su2double limits_table_prog[2];
+
+  /* !brief
+   * Holds the variable names stored in the table file.
+   * Order is in sync with data
+   */
+  vector<string> names_var;
+
+  /* !brief
+   * Holds all data stored in the table. First index addresses the variable
+   * while second index addresses the point.
+   */
+  su2activematrix table_data;
+
+  su2matrix<unsigned long> triangles;
+
+  /* we do not know this size in advance until we go through the entire lookup table */
+  vector<vector<unsigned long> > edges;
+  vector<vector<unsigned long> > edge_to_triangle;
+
+  vector<unsigned long> hull;
+
+  CTrapezoidalMap trap_map_prog_enth;
+
+  vector<su2activematrix> interp_mat_inv_prog_enth;
+
+  inline int GetIndexOfVar(string nameVar) {
+    int index = find(names_var.begin(), names_var.end(), nameVar) - names_var.begin();
+    if (index == int(names_var.size())) {
+      index = -1;
+      string error_msg = "Variable '";
+      error_msg.append(nameVar);
+      error_msg.append("' is not in the lookup table.");
+      SU2_MPI::Error(error_msg, CURRENT_FUNCTION);
+    }
+    return index;
+  }
+
+  inline const vector<su2double> GetData(string name_var) {
```

Comment:
Pass strings by const&, and please add the required `std::` and do not use `using namespace std;` at least in hpp files.
```suggestion
  inline const su2double* GetData(const std::string& name_var) const {
```